dashboard/app/models/script.rb

```diff
@@ -1125,55 +1125,63 @@ def fix_lesson_positions
   end
 
   def clone_migrated_unit(new_name, new_level_suffix: nil, destination_unit_group_name: nil, version_year: nil, family_name:  nil)
+    raise 'Script name has already been taken' if Script.find_by_name(new_name)
```

Comment:
in an abundance of caution, I would suggest checking for the existence of this file (not just the db object) and fail in that case too.